src/runtime/lpython_builtin.py

```diff
@@ -1085,6 +1085,65 @@ def _lpython_str_isspace(s: str) -> bool:
             return False
     return True
 
+@overload
+def _lpython_str_center(s: str, width: i32, fillchar: str) -> str:
+    """
+    Return centered in a string of length width. 
+    Padding is done using the specified fillchar (default is an ASCII space). 
+    The original string is returned if width is less than or equal to len(s).
+    """
+    if(len(fillchar) != 1):
+        raise TypeError("The fill character must be exactly one character long")
+    if(width <= len(s)):
+        return s
+    width -= len(s)
+    result: str = ""
+    left_padding: i32
+    if width/2 > f64(i32(width/2)):
+        left_padding = i32(width/2) + 1
+    else:
+        left_padding = i32(width/2)
+    i: i32 
+    for i in range(left_padding):
+        result += fillchar
+    right_padding: i32 = width - left_padding
+    result += s
+    for i in range(right_padding):
+        result += fillchar
```

Comment:
We can utilize the new improvement in string repeat added a dew days ago and simplify this to just:
```python
right_padding: i32 = width - left_padding
result += fillchar * left_padding
result += s
result += fillchar * right_padding
```

#2652 needs to be addressed to achieve it. For now we may keep the current `for` loop.